runtime/util/hshelp.c

```diff
@@ -2641,6 +2642,15 @@ recreateRAMClasses(J9VMThread * currentThread, J9HashTable * classHashTable, J9H
 		/* Delete original class from defining loader's class table */
 		if (!fastHCR) {
 			vmFuncs->hashClassTableDelete(classLoader, J9UTF8_DATA(className), J9UTF8_LENGTH(className));
+			if ((NULL != vm->sharedClassConfig) && (NULL != vm->sharedClassConfig)
+				&& J9_ARE_ALL_BITS_SET(vm->sharedCacheAPI->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)
```

Comment:
This should be `(NULL != vm->sharedClassConfig) && (NULL != vm->sharedClassConfig->romToRamHashTable)`. No need to check `J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES`.